### PR TITLE
Driver for distance sensor: vl53l1x

### PR DESCRIPTION
**Pull request by shizacat**
Hello.
This is driver for distance sensor VL53L1x from ST.  Based on the Ultra lite driver, UM2510; [en.STSW-IMG009](https://www.st.com/en/embedded-software/stsw-img009.html).

### CODE REVIEW

**Comment by dagar** [comment]
Looks good, any test data you can share? I'm wondering about the default ranging mode. 

Other than that the only thing I'm wondering is it we should keep it closer to the form of the ST Ultra lite driver for future updates, unless it's quite unlikely they'll update it.

File: src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+        perf_count(_comms_errors);
+        return ret;
+    }
+
+    *data = ((uint16_t)data_out[0]<<8) + (uint16_t)data_out[1];
+
+    return PX4_OK;
+}
+
+// int8_t VL53L1X::VL53L1_RdDWord(uint16_t index, uint32_t *data) {
```
Comment (dagar) [inline src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp:318]: Are these going to be needed later? If not, please delete all the dead commented code.

File: src/drivers/distance_sensor/vl53l1x/vl53l1x.hpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2018-2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/distance_sensor/vl53l1x/vl53l1x.hpp:3]:
```suggestion
 *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp [hunk 3]
```diff
@@ -0,0 +1,741 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2018 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp:3]:
```suggestion
 *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/distance_sensor/vl53l1x/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,43 @@
+############################################################################
+#
+#   Copyright (c) 2019 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/distance_sensor/vl53l1x/CMakeLists.txt:3]:
```suggestion
#   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```

File: src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp (hunk 1 above)
Comment (shizacat) [reply src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp]: I thought it might be useful for the ability to add some functions from the original driver

**Comment by bys1123** [comment]
Looking forward vl53l1x get supported.

**Comment by shizacat** [comment]
Very soon I will start to test. The propellers were broken. Later I forgotten.... sorry

**Comment by shizacat** [comment]
This logs: https://logs.px4.io/plot_app?log=93bd63e7-9c90-4e88-8cee-6853287f1e47

Not sure, what every things works fine, but data from sensor has (distance sensor plot)

**Comment by dagar** [comment]
> This logs: https://logs.px4.io/plot_app?log=93bd63e7-9c90-4e88-8cee-6853287f1e47
> 
> Not sure, what every things works fine, but data from sensor has (distance sensor plot)

Can you try to cleanup the branch, perhaps squash all your changes into a single commit? It's not currently mergable or really even reviewable.

**Pull request closed without merging**